src/Menu/Builder.php

```diff
@@ -5,6 +5,7 @@
 
 
 use Illuminate\Contracts\Routing\UrlGenerator;
+use Illuminate\Support\Facades\Gate;
```

Comment:
Can you inject the Gate in the constructor instead of using the Facade? This makes it better testable.